sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -638,6 +641,18 @@ object GpuCast {
     }
   }
 
+  private def fixupToNumException[A](input: ColumnView, to: DataType)(f: ColumnView => A): A = {
+    try {
+      f(input)
+    } catch {
+      case c: CastException =>
+        val s = withResource(input.copyToHost()) { hcv =>
```

Comment:
One concern: `copyToHost` will copy all the rows and will contribute to peak memory.
It's in an exception status, so I do not know whether it's necessary to improve the memory usage, e.g.: develop a JNI API to only copy one row into CPU.

Reply:
This is a good suggestion and an improvement, we can do this as a follow up. And we may need this improvement for many places. e.g.
https://github.com/NVIDIA/spark-rapids/blob/branch-25.08/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/mathExpressions.scala#L822
https://github.com/NVIDIA/spark-rapids/blob/branch-25.08/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala#L1447

Reply:
Filed https://github.com/NVIDIA/spark-rapids-jni/issues/3492

Reply:
https://github.com/NVIDIA/spark-rapids/pull/13046 will address this.